sw/device/lib/testing/test_main.h

```diff
@@ -12,6 +12,41 @@
  * @brief Entrypoint definitions for on-device tests
  */
 
+/**
+ * Configuration variables for an on-device test.
+ *
+ * This type represents configuration values for an on-device test, which allow
+ * tests to cofigure the behavior of the `test_main.h` harness.
+ *
+ * New fields can be safely added to this struct without affecting any tests;
+ * the "default" value of all fields should be zero (or NULL, or equivalent).
+ *
+ * See `kTestConfig`.
+ */
+typedef struct test_config {
+  /**
+   * Indicates that `test_main()` does something non-trivial to the UART
+   * device. Setting this to true will make `test_main()` guard against this
+   * by resetting the UART device before printing debug information.
+   */
+  bool can_clobber_uart;
```

Comment:
I think this might be more accurate as `will_clobber_uart`. What do you think?

Reply:
I originally thought about naming that too. I kept "can" instead, since "will" sort of implies a guarantee.

That said, I'm debating abandoning this PR entirely and instead making *all* tests reinitialize UART and PLIC and whatnot when the code under test ends; I'm thinking of even injecting this into CHECK through some atexit-like mechanism. What do you think?

Reply:
I don't think we don't need to get more complex than this PR, and probably re-initializing the UART and the PLIC in every test is silly (especially when DV doesn't even use UART). 

Today, let's fix what @sriyerg asked and just merge this, rather than building the world's most fully-featured and perfect testing system.